spring-cloud-aws-autoconfigure/src/main/java/org/springframework/cloud/aws/autoconfigure/context/ContextCredentialsAutoConfiguration.java

```diff
@@ -16,100 +16,69 @@
-import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
 import org.springframework.cloud.aws.autoconfigure.context.properties.AwsCredentialsProperties;
-import org.springframework.cloud.aws.context.config.annotation.ContextDefaultConfigurationRegistrar;
-import org.springframework.cloud.aws.core.credentials.CredentialsProviderFactoryBean;
-import org.springframework.context.EnvironmentAware;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
-import org.springframework.context.annotation.Import;
-import org.springframework.context.annotation.ImportBeanDefinitionRegistrar;
-import org.springframework.core.env.Environment;
-import org.springframework.core.type.AnnotationMetadata;
 import org.springframework.util.StringUtils;
 
-import static com.amazonaws.auth.profile.internal.AwsProfileNameLoader.DEFAULT_PROFILE_NAME;
-import static org.springframework.cloud.aws.context.config.support.ContextConfigurationUtils.registerCredentialsProvider;
-import static org.springframework.cloud.aws.context.config.support.ContextConfigurationUtils.registerDefaultAWSCredentialsProvider;
+import static org.springframework.cloud.aws.core.config.AmazonWebserviceClientConfigurationUtils.CREDENTIALS_PROVIDER_BEAN_NAME;
 
 /**
+ * {@link EnableAutoConfiguration} for {@link AWSCredentialsProvider}.
+ *
  * @author Agim Emruli
+ * @author Maciej Walkowiak
  */
 @Configuration(proxyBeanMethods = false)
-@Import({ ContextDefaultConfigurationRegistrar.class,
-		ContextCredentialsAutoConfiguration.Registrar.class })
-@ConditionalOnClass(name = "com.amazonaws.auth.AWSCredentialsProvider")
+@EnableConfigurationProperties(AwsCredentialsProperties.class)
+@ConditionalOnClass(com.amazonaws.auth.AWSCredentialsProvider.class)
 public class ContextCredentialsAutoConfiguration {
 
-	/**
-	 * The prefix used for AWS credentials related properties.
-	 */
-	public static final String AWS_CREDENTIALS_PROPERTY_PREFIX = "cloud.aws.credentials";
-
-	/**
-	 * Bind AWS credentials related properties to a property instance.
-	 * @return An {@link AwsCredentialsProperties} instance
-	 */
-	@Bean
-	@ConfigurationProperties(prefix = AWS_CREDENTIALS_PROPERTY_PREFIX)
-	public AwsCredentialsProperties awsCredentialsProperties() {
-		return new AwsCredentialsProperties();
-	}
+	@Bean(name = CREDENTIALS_PROVIDER_BEAN_NAME)
+	@ConditionalOnMissingBean(name = CREDENTIALS_PROVIDER_BEAN_NAME)
+	public AWSCredentialsProvider awsCredentialsProvider(
+			AwsCredentialsProperties properties) {
 
-	/**
-	 * Registrar for the credentials provider.
-	 */
-	public static class Registrar
-			implements ImportBeanDefinitionRegistrar, EnvironmentAware {
+		List<AWSCredentialsProvider> providers = new ArrayList<>();
```

Comment:
what about having this `List<AWSCredentialsProvider> providers = resolveCredentialsProvider(properties);` and the method `private List<AWSCredentialsProvider> resolveCredentialsProvider(properties)` contains all the logic

Reply:
👍